### DISCUSSION TITLE
Why doesn't in operator type guard work with Sequence ?

### DISCUSSION THREAD

**Discussion by robbi** [microsoft/pyright#9621] [Q&A]
As mentioned in https://github.com/microsoft/pyright/blob/main/docs/type-concepts-advanced.md#type-guards type guards `in` and `not in` works for list, set, frozenset, deque, tuple, dict, defaultdict, and OrderedDict.
Why is `Sequence` not supported ?

**Comment by erictraut** [comment]
I don't think `Sequence` guarantees the semantics for `in` that are required for safe type narrowing. The other classes in this list have well-documented and guaranteed behaviors in this regard.
